test/e2e/tls/tls.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2019 Intel Corporation.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package tls
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+
+	pmempod "github.com/intel/pmem-csi/test/e2e/pod"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/format"
+)
+
+var _ = Describe("TLS", func() {
+	f := framework.NewDefaultFramework("tls")
+
+	// All of the following pod names, namespaces and ports match
+	// those in the current deployment files.
+
+	var nodePod *v1.Pod
+	BeforeEach(func() {
+		// Find one node driver pod.
+		label := labels.SelectorFromSet(labels.Set(map[string]string{"app": "pmem-csi-node"}))
+		pods, err := f.ClientSet.CoreV1().Pods("default").List(metav1.ListOptions{LabelSelector: label.String()})
+		framework.ExpectNoError(err, "list PMEM-CSI node pods")
+		Expect(pods.Items).NotTo(BeEmpty(), "have PMEM-CSI node pods")
+		nodePod = &pods.Items[0]
+	})
+
+	Context("controller", func() {
+		It("is secure", func() {
+			checkTLS(f, "pmem-csi-controller-0.pmem-csi-controller.default")
+		})
+	})
+	Context("node", func() {
+		It("is secure", func() {
+			checkTLS(f, nodePod.Status.PodIP)
+		})
+	})
+})
+
+func checkTLS(f *framework.Framework, server string) {
+	containerName := "nmap"
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      containerName,
+			Namespace: f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    "nmap",
+					Image:   os.Getenv("PMEM_CSI_IMAGE"),
```

Comment:
@pohly So, this expect pmem-csi image to have `nmap` installed?

Reply:
Ok, Got it. The test installs the bundle here: https://github.com/intel/pmem-csi/pull/545/commits/a2ee870293a82b94efd3ef55bfcbc85af9b697df#diff-d5ba72216cc5edc8c882d8d767bbad46R90